src/session.rs

```diff
@@ -32,6 +35,49 @@ impl Session {
         }
     }
 
+    /// Loads a session from an exported model.
+    pub fn from_saved_model<P: AsRef<Path>, Tag: AsRef<str>, Tags: IntoIterator<Item = Tag>>
+        (options: &SessionOptions,
+         tags: Tags,
+         graph: &mut Graph,
+         export_dir: P)
+         -> Result<Self> {
+        let mut status = Status::new();
+
+        let export_dir_cstr = try!(export_dir.as_ref()
+                                       .to_str()
+                                       .and_then(|s| CString::new(s.as_bytes()).ok())
+                                       .ok_or_else(|| {
+            Status::new_set(Code::InvalidArgument, "Invalid export directory path").unwrap()
+        }));
+
+        let tags_cstr: Vec<_> = try!(tags.into_iter()
+                                         .map(|t| CString::new(t.as_ref()))
+                                         .collect::<StdResult<_, _>>()
+                                         .map_err(|_| {
+            Status::new_set(Code::InvalidArgument, "Invalid tag name").unwrap()
+        }));
+        // keeping tags_cstr to retain strings in memory
+        let tags_ptr: Vec<*const c_char> = tags_cstr.iter().map(|t| t.as_ptr()).collect();
+
+        let inner = unsafe {
+            tf::TF_LoadSessionFromSavedModel(options.inner,
+                                             ptr::null(),
+                                             export_dir_cstr.to_bytes_with_nul().as_ptr() as
```

Comment:
The return value of `to_bytes_with_nul` doesn't live long enough.  You can just use `export_dir_cstr.as_ptr()`, since it already guarantees a null terminator.